tests/mbt/driver/mbt_test.go

```diff
@@ -0,0 +1,577 @@
+		actualRunningConsumers := make([]string, len(systemRunningConsumers))
+		for i, chain := range systemRunningConsumers {
+			actualRunningConsumers[i] = chain.ChainId
+		}
+
+		// sort the slices so that we can compare them
+		sort.Slice(modelRunningConsumers, func(i, j int) bool {
+			return modelRunningConsumers[i] < modelRunningConsumers[j]
+		})
+		sort.Slice(actualRunningConsumers, func(i, j int) bool {
+			return actualRunningConsumers[i] < actualRunningConsumers[j]
+		})
+
+		require.Equal(t, modelRunningConsumers, actualRunningConsumers, "Running consumers do not match")
+
+		// check validator sets - provider current validator set should be the one from the staking keeper
+		CompareValidatorSets(t, driver, currentModelState, actualRunningConsumers, index)
+
+		// check times - sanity check that the block times match the ones from the model
+		CompareTimes(driver, actualRunningConsumers, currentModelState, timeOffset)
+
+		// check sent packets: we check that the package queues in the model and the system have the same length.
+		for _, consumer := range actualRunningConsumers {
+			ComparePacketQueues(t, driver, currentModelState, consumer, timeOffset)
+		}
+		// compare that the sent packets on the proider match the model
+		CompareSentPacketsOnProvider(driver, currentModelState, timeOffset)
+
+		stats.EnterStats(driver)
+
+		t.Logf("State %v of trace %v is ok!", index, path)
+	}
+	t.Log("🟢 Trace is ok!")
+}
+
+func CompareValidatorSets(t *testing.T, driver *Driver, currentModelState map[string]itf.Expr, consumers []string, index int) {
+	t.Helper()
+	modelValSet := ValidatorSet(currentModelState, "provider")
+	curValSet := driver.providerValidatorSet()
+
+	actualValSet := make(map[string]int64, len(curValSet))
+
+	for _, val := range curValSet {
+		valName := val.Description.Moniker
+		actualValSet[valName] = val.Tokens.Int64()
+	}
+
+	require.NoError(t, CompareValSet(modelValSet, actualValSet), "Validator sets do not match")
+
+	for _, consumer := range consumers {
+		modelValSet = HistoricalValidatorSet(currentModelState, consumer, 0)
+
+		consumerVals := driver.consumerValidatorSet(ChainId(consumer))
+		consumerCurValSet := make(map[string]int64, len(consumerVals))
+		for _, val := range consumerVals {
+			pubkey, err := val.ConsPubKey()
+			require.NoError(t, err, "Error getting pubkey")
+
+			consAddr := providertypes.NewConsumerConsAddress(sdktypes.ConsAddress(pubkey.Address().Bytes()))
+
+			providerConsAddr, found := driver.providerKeeper().GetValidatorByConsumerAddr(driver.providerCtx(), consumer, consAddr)
+			if !found {
+				providerConsAddr = providertypes.NewProviderConsAddress(consAddr.Address)
+			}
+
+			providerVal, found := driver.providerStakingKeeper().GetValidatorByConsAddr(driver.providerCtx(), providerConsAddr.Address)
+			require.True(t, found, "Error getting provider validator")
+
+			consumerCurValSet[providerVal.GetMoniker()] = val.Power
+		}
+		for val, power := range modelValSet {
```

Comment:
Is this `for` loop needed?

Reply:
Woops! Yup, that indeed does not do anything anymore